app/src/main/java/org/mozilla/fenix/historymetadata/controller/HistoryMetadataController.kt

```diff
@@ -53,6 +62,13 @@ class DefaultHistoryMetadataController(
         )
     }
 
+    override fun handleRemoveGroups(groups: Set<HistoryMetadataGroup>) {
+        activity.components.core.historyMetadataService.removeMetadata(groups)
```

Comment:
This method should take in a single `group`, and can just do a simple storage call: `activity.components.core.historyStorage.deleteHistoryMetadata(group.title)`

Internally this will remove all of metadata in the places database that matches this `searchTerm`. So, it's not necessary to iterate over what's in the group.